src/shared/components/DateWidget.tsx

```diff
@@ -3,27 +3,36 @@ import { TextFieldProps } from "@material-ui/core";
 import MuiDatePicker from "@material-ui/lab/DatePicker";
 import AdapterDateFns from "@material-ui/lab/AdapterDateFns";
 import LocalizationProvider from "@material-ui/lab/LocalizationProvider";
-import { format, isValid } from "date-fns";
+import { parse, format, isValid } from "date-fns";
 import { WidgetProps } from "@rjsf/core";
 import { TextWidget } from "./TextWidget";
 import enLocale from "date-fns/locale/en-NZ";
 
 // TODO: Fix typings for TextWidget
 export const DateWidget: FC<WidgetProps> = (props: any) => {
-  const [selectedDate, setSelectedDate] = useState<Date | null>(null);
+  const { label, schema, value } = props;
   const handleDateChange = (date: Date | null) => {
     if (isValid(date)) {
       const validDate = date || new Date();
       props.onChange(format(validDate, "dd/MM/yyyy"));
-      setSelectedDate(validDate);
+    }
+  };
+
+  const getDateValue = () => {
+    const asDate = parse(value, "dd/mm/yyyy", new Date());
+
+    if (isValid(asDate)) {
+      return parse(value, "dd/mm/yyyy", new Date());
+    } else {
+      return null;
```

Comment:
```suggestion
    return isValid(asDate) ? asDate : null;
```

no need to parse again, I think. in which case..